components/ipfs/ipfs_p3a.cc

```diff
@@ -46,44 +46,86 @@ int GetDaemonUsageBucket(base::TimeDelta elapsed_time) {
 void RecordIPFSDetectionPromptCount(PrefService* prefs) {
   const int max_bucket = 4;
   int bucket = GetIPFSDetectionPromptBucket(prefs);
-  UMA_HISTOGRAM_EXACT_LINEAR("Brave.IPFS.DetectionPromptCount", bucket,
+  UMA_HISTOGRAM_EXACT_LINEAR(kDetectionPromptCountHistogramName, bucket,
                              max_bucket);
 }
 
 // IPFS state
 // Ask (0), Gateway (1), Local Node (2), Disabled (3)
 void RecordIPFSGatewaySetting(PrefService* prefs) {
   auto resolve_method = prefs->GetInteger(kIPFSResolveMethod);
-  UMA_HISTOGRAM_EXACT_LINEAR("Brave.IPFS.GatewaySetting", resolve_method, 4);
+  UMA_HISTOGRAM_EXACT_LINEAR(kGatewaySettingHistogramName, resolve_method, 4);
+}
+
+// Was the IPFS local node installed? If so, is it still used?
+// Never installed (0), Installed (1), Installed, not used any more (2)
+void RecordIPFSLocalNodeRetention(PrefService* prefs) {
+  auto resolve_method = static_cast<IPFSResolveMethodTypes>(
+      prefs->GetInteger(kIPFSResolveMethod));
+  auto local_node_used = prefs->GetBoolean(kIPFSLocalNodeUsed);
+  int bucket = 0;
+  switch (resolve_method) {
+    case IPFSResolveMethodTypes::IPFS_ASK:
+      bucket = local_node_used;
+      break;
+    case IPFSResolveMethodTypes::IPFS_LOCAL:
+      if (!local_node_used)
+        prefs->SetBoolean(kIPFSLocalNodeUsed, true);
+      bucket = 1;
+      break;
+    case IPFSResolveMethodTypes::IPFS_GATEWAY:
+    case IPFSResolveMethodTypes::IPFS_DISABLED:
+      if (local_node_used)
+        bucket = 2;
+      break;
+    default:
+      break;
+  }
+  UMA_HISTOGRAM_EXACT_LINEAR(kLocalNodeRetentionHistogramName, bucket, 3);
```

Comment:
3 -> 2

Reply:
It appears that we may have found another inconsistency with using "exclusive max" vs "real max" values in the histogram macro. In this source file, you will see that the other macro usages use the exclusive max unlike what we saw with Rewards P3A. We might need to comb through the code base and update each `UMA_HISTOGRAM_EXACT_LINEAR` usage to use either the exclusive or real max, at some point. My hunch is that we should use exclusive max for all usages, since that's what's intended by Chromium.

I can update this usage to use the real max (2), along with all other usages in this file for now. Please advise, thx

cc @bbondy

Reply:
i think the better way would be to make our own tools for recording p3a metrics that would wrap UMA and not confuse people with exclusive/real max. Let me file an issue :) 
I guess we can leave this one alone for now